### ISSUE TITLE
Reporting the cursor position requires pressing enter?

### ISSUE THREAD

**Issue by nhooyr** [neovim/neovim#4151]
Very simple and easy to reproduce. 

First open a normal terminal and run `echo -e "\033[6n" && cat`

You should see something like `^[[58;1R`

Thats great, its reporting the cursor position. Try the same in a neovim terminal. No reporting until you press enter for some reason.

**Comment by nhooyr** [comment]
@justinmk shouldn't this be labeled a bug? its very simple to reproduce.

**Comment by justinmk** [comment]
@leonerd Is this expected behavior in libvterm?

**Comment by leonerd** [comment]
No. I suspect it might be the result of `nvim` not correctly applying the required termios fiddlings to the pty/tty device pair it created. Or maybe somewhere else in code that isn't flushing bytes in/out in a timely fashion

**Comment by sisrfeng** [comment]
How to map `<C-r>`  to  zsh' s keybind in terminal mode? 

I have this in my .zshrc:
bindkey '\C-r' history-peco

history-peco is :

![image](https://user-images.githubusercontent.com/53520949/153225882-790a4858-8d54-4885-9f98-17141df1eeec.png)

With  `tnoremap <c-r><c-r> <c-r>" `,  if I press `<c-r><c-r>`, I get my zsh 's ctrl-r back instead of pasting from buffer `"`  

what' s happening?

Thx!
